bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/ApiBridgeHandler.java

```diff
@@ -348,27 +350,26 @@ public synchronized <T> T executeUri(URI uri, HttpMethod method, Class<T> clazz,
             try {
                 exception = new NetatmoException(deserializer.deserialize(ApiError.class, responseBody));
             } catch (NetatmoException e) {
-                exception = new NetatmoException("Error deserializing error: %s".formatted(statusCode.getMessage()));
+                if (statusCode == Code.TOO_MANY_REQUESTS) {
+                    exception = new NetatmoException(statusCode.getMessage());
+                } else {
+                    exception = new NetatmoException(
+                            "Error deserializing error: %s".formatted(statusCode.getMessage()));
+                }
             }
-            throw exception;
-        } catch (NetatmoException e) {
-            if (e.getStatusCode() == ServiceError.MAXIMUM_USAGE_REACHED) {
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
-                prepareReconnection(null, null);
+            if (statusCode == Code.TOO_MANY_REQUESTS
+                    || exception.getStatusCode() == ServiceError.MAXIMUM_USAGE_REACHED) {
+                prepareReconnection(API_LIMIT_INTERVAL_S,
+                        "@text/maximum-usage-reached [ \"%d\" ]".formatted(API_LIMIT_INTERVAL_S), null, null);
             }
-            throw e;
-        } catch (InterruptedException e) {
-            Thread.currentThread().interrupt();
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
-            throw new NetatmoException("Request interrupted");
-        } catch (TimeoutException | ExecutionException e) {
+            throw exception;
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
```

Comment:
I'm not sure the `InterruptedException` handling is correct, but I think it was not correct previously either because the handler probably didn't respect it after being converted to `NetatmoException`. This might give a bit insight: https://rules.sonarsource.com/java/RSPEC-2142/

By initiating a retry below, it seems we are delaying the thread interruption. Perhaps it would be better to propagate the exception?

@wborn - can you advise a bit here?

Reply:
When the `InterruptedException` is catched, it should set the interrupt flag again.
`Thread.currentThread().interrupt();`

Also the rescheduling should be prevented, the thread should exit a.s.a.p.

Reply:
Reviewed handling of InterruptedException according to you observation @lsiepel